docs/generation_metageneration.rst

```diff
@@ -0,0 +1,114 @@
+Conditional Requests Via Generation / Metageneration
```

Comment:
I'm thinking maybe we can incorporate "Preconditions" into the title for consistency. WDYT? We can also add an intro paragraph similar to the [doc](https://cloud.google.com/storage/docs/generations-preconditions#_Preconditions), something like: 

"Preconditions tell Cloud Storage to only perform a request if the generation or metageneration number of the affected object meets your precondition criteria. These checks of the generation and metageneration numbers ensure that the object is in the expected state, allowing you to perform safe read-modify-write updates and conditional operations on objects."

Reply:
b9367f4